### PR TITLE
fix: add sbom to the release process

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Include Attach backend/frontend  SBOM to the release artefacts through the release job published event

- **`on.release.types: [published]`**: This triggers the workflow when a release is published on GitHub.
- **`Attach SBOM to release` step**: This step uses `actions/upload-release-asset@v1` to upload `bom.json` as an asset to the release. It only runs if the event that triggered the workflow is a release, as specified by the condition `if: github.event_name == 'release'`.

Action test:
https://github.com/camunda/camunda/actions/runs/12012595948

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Review by clementnero — COMMENTED** [review]
LGTM!  Regarding potential regression, there is no concern since `cyclonedx` plugin is not configured by default.
Looking at the release aspect, I would suggest to request  a review from @camunda/monorepo-ci-dri

File: .github/workflows/sbom.yml
Comment (nathansandi) [reply .github/workflows/sbom.yml:45]: SBOM generation often involves accurate dependency resolution.
Using cached data could potentially introduce discrepancies if the cache is stale or incomplete. Caching might bypass fetching the latest versions, leading to outdated SBOMs.

File: .github/workflows/sbom.yml
Comment (nathansandi) [reply .github/workflows/sbom.yml]: I reintroduce this :) It was removed because I was using a draft of a wrong version on my machine

**Comment by nathansandi** [comment]
> @nathansandi since the 8.7.0-alpha2 release is currently ongoing, do you already know when you want to merge this? Maybe coordinate/double-check with the release manager in Slack since adding the new `on.release.types: [published]` trigger could maybe interfere with the ongoing release.

Nice point @cmur, I am going to ping the release manager, it would be great opportunity to test the whole workflow in a nutshell.

**Comment by nathansandi** [comment]
@cmur2  It wont be possible run it on this release, once the release train was already dispatched.
I would like to merge this in order to test in the next one. Please let me know if any change is need here (if necessary). Considering it has no approval yet, Happy to help in case necessary.  👍

**Review by cmur2 — APPROVED** [review]

**Pull request merged**
